tests/testsuite/rustflags.rs

```diff
@@ -690,6 +731,25 @@ fn build_rustflags_normal_source_with_target() {
         .with_status(101)
         .with_stderr_contains("[..]bogus[..]")
         .run();
+
+    // With -Ztarget-applies-to-host build.rustflags should apply to build.rs since target == host
```

Comment:
The `target-applies-to-host` flag was desgined to control the behavior of `target` configs, not `build` configs, for example it should determine if `target.<triple>.rustflags` is applied to `build.rs`. The `target-applies-to-host` config flag was not intended to change the behavior of non-target configs like `build.rustflags`.

Reply:
I know that it wasn't intended to, but I think it's still the right way for it to behave. It's just that the behavior was inconsistent before. The [docs for `build.rustflags`](https://doc.rust-lang.org/nightly/cargo/reference/config.html#buildrustflags) say:

> If the `--target` flag (or [`build.target`](https://doc.rust-lang.org/nightly/cargo/reference/config.html#buildtarget)) is used, then the flags will only be passed to the compiler for the target.

If `target-applies-to-host = true`, that implies that the same rules should be used for host artifacts as are used for `--target <host>`. Which would include `build.rustflags` _if_ `target.<host>` is unset.

That said, the immediately following sentence of the docs would need to be updated:

> Things being built for the host, such as build scripts or proc macros, will not receive the args.

Reply:
> If `target-applies-to-host = true`, that implies that the same rules should be used for host artifacts as are used for `--target <host>`. Which would include `build.rustflags` _if_ `target.<host>` is unset.

Sure, that's probably technically more correct, but the purpose of setting `target-applies-to-host = true` was for backwards/forwards compatibility while migrating the defaults and not really something intended to be used unless needed for compatibility purposes. The idea is that the `target-applies-to-host = true` config is basically a null-op until the defaults are changed which could be pre-emptively set by projects so that the default change would not affect that project.